plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -118,6 +118,7 @@ public TransactionLogAccess(
                 .build();
         activeDataFileCache = EvictableCacheBuilder.newBuilder()
                 .weigher((Weigher<String, DeltaLakeDataFileCacheEntry>) (key, value) -> Ints.saturatedCast(estimatedSizeOf(key) + value.getRetainedSizeInBytes()))
+                .expireAfterWrite(deltaLakeConfig.getDataFileCacheTtl().toMillis(), TimeUnit.MILLISECONDS)
```

Comment:
I guess we should add `shareNothingWhenDisabled()`. Also, could you add a test with `delta.metadata.live-files.cache-ttl=0s` in `TestDeltaLakePlugin`?

Reply:
Good point. I missed adding  `shareNothingWhenDisabled()`.